### PR TITLE
Context matches v1 type.

### PR DESCRIPTION
**Pull request by tcnichol**
In clowder v1 context was a list that consisted in URL and dictionary. This pull request makes v2 match v1. Front end and back end have both been changed. Extractor registration should work correctly.

If testing, make sure to remove old metadata/listeners from DB, or else conflicts will happen.

### CODE REVIEW

**Comment by longshuicy** [comment]
Can you fix the linting?

**Comment by tcnichol** [comment]
To test with extractors, use this branch of pyclowder:

https://github.com/clowder-framework/pyclowder/pull/60

Wordcount and other sample extractors there can be used.

**Comment by tcnichol** [comment]
For the errors in pytest. These are happening when metadata with no context_url is being posted to the backend for files and datasets. The field is an optional[str] but the validator looks like it throws an error if it is missing. Not sure what to do to fix it. Also that the changes made in this PR shouldn't affect that.

**Comment by tcnichol** [comment]
@ddey2 I pushed a small change that might fix the errors you are seeing.

**Review by ddey2 — APPROVED** [review]
Looks good, tested.

File: frontend/src/components/metadata/CreateMetadataDefinition.tsx [hunk 1]
```diff
@@ -118,10 +118,12 @@ export const CreateMetadataDefinition = (): JSX.Element => {
             contextJson[item["term"]] = item["iri"]
         })
 
+		console.log([JSON.stringify(contextJson)]);
+
```
Comment (max-zilla) [inline frontend/src/components/metadata/CreateMetadataDefinition.tsx:122]: please remove

**Review by max-zilla — CHANGES_REQUESTED** [review]
everything seemed to work, remove the log and i will merge after tests pass thanks

File: frontend/src/components/metadata/CreateMetadataDefinition.tsx (hunk 1 above)
Comment (tcnichol) [reply frontend/src/components/metadata/CreateMetadataDefinition.tsx]: removed

**Pull request merged**